src/os/bluestore/BlueStore.cc

```diff
@@ -6045,10 +6045,42 @@ void BlueStore::_txc_state_proc(TransContext *txc)
       _txc_finalize_kv(txc, txc->t);
       if (!sync_transaction) {
 	if (sync_submit_transaction) {
+	  std::lock_guard<std::mutex> l(id_lock);
```

Comment:
@liewegas . I review the code and found how to update nid_max/blobid_max for parallel submit-txn
a)before update oid/blobkid of txn(which larger than *_max) it must firstly update *_max
b)don't occur this problem: t1->max_nid > t2->max_nid, But the submit order is t1, t2.
I firstly use the following code:
__assign_nid {
if (nid_last > nid_max && (parallel_submit_txn)
   wait(kv_sync_thread update *_max).
}
The other is this commit. I don't know which is better.  Hope your comments! Thanks!